src/stan/services/pathfinder/multi.hpp

```diff
@@ -0,0 +1,210 @@
+#ifndef STAN_SERVICES_PATHFINDER_MULTI_HPP
+#define STAN_SERVICES_PATHFINDER_MULTI_HPP
+
+#include <stan/callbacks/interrupt.hpp>
+#include <stan/callbacks/logger.hpp>
+#include <stan/callbacks/writer.hpp>
+#include <stan/io/var_context.hpp>
+#include <stan/optimization/bfgs.hpp>
+#include <stan/optimization/lbfgs_update.hpp>
+#include <stan/services/pathfinder/single.hpp>
+#include <stan/services/pathfinder/psis.hpp>
+#include <stan/services/error_codes.hpp>
+#include <stan/services/util/initialize.hpp>
+#include <stan/services/util/create_rng.hpp>
+#include <tbb/parallel_for.h>
+#include <tbb/concurrent_vector.h>
+#include <boost/random/discrete_distribution.hpp>
+#include <boost/iterator.hpp>
+#include <string>
+#include <vector>
+
+namespace stan {
+namespace services {
+namespace pathfinder {
+
+template <typename T>
+double duration_seconds(const T& start, const T& end) {
+  return std::chrono::duration_cast<std::chrono::milliseconds>(end - start)
+             .count()
+         / 1000.0;
+}
+
+/**
+ * Runs multiple pathfinders with final approximate samples drawn using PSIS.
+ *
+ * @tparam Model A model implementation
+ * @tparam InitContext Type inheriting from `stan::io::var_context`
+ * @tparam InitWriter Type inheriting from `stan::io::var_context`
```

Comment:
isn't this type inheriting from stan::callbacks::writer?

Reply:
👍 fixing